src/Components/ProfilePageContext.js

```diff
@@ -10,7 +10,8 @@ const ProfilePageContext = createContext({
   updateAvatar: (avatar) => {},
   updateLikes: (likes) => {},
   updateDislikes: (dislikes) => {},
-  updateList: (index, list) => {},
+  updateList: (list) => {},
+  updateSavedList: (userId, listId) => {},
```

Comment:
Just saw github must have dropped my big comment 😡.  Here is the gist:

`updateList` kinda makes sense here but kinda does not -- it does edit your LocalUser, but it is not like editing your profile since it actually happens on other people's pages.  It is kinda like "liking" a show or adding it to a list.  In that spirit, I propose the code to save/unsave a list gets a hook similar to `useLikeState`.  Maybe:
```js
export default function useListSavedState(listId, userId) {
  const [user] = useAuthState(auth);
  const [localUser, setLocalUser] = useContext(LocalUserContext);

  const savedLists = localUser?.savedLists ?? [];

  const entry = { listId, userId };

  const saved = Boolean(savedLists.find((x) => matchesEntry(x, entry)));

  const setSaved = (value) => {
    const newLocalUser = {
      ...localUser,
      savedLists: value ? add(savedLists, entry) : remove(savedLists, entry),
    };

    setLocalUser(newLocalUser);
    SaveToFirestore(user, newLocalUser);
  };

  return {
    saved,
    setSaved,
  };
}

// Equality matcher for two savedLists entries.
function matchesEntry(a, b) {
  return a.listId === b.listId && a.userId === b.userId;
}

// Adds `entry` to `savedLists`, if not already present.  Always returns a new list.
function add(savedLists, entry) {
  if (savedLists.find((x) => matchesEntry(x, entry))) {
    return [...savedLists];
  } else {
    return [...savedLists, entry];
  }
}

// Removes an `entry` from `savedLists` if it is present. Always returns a new list.
function remove(savedLists, entry) {
  return savedLists.filter((x) => !matchesEntry(x, entry));
}
```

And then it could be used like:
```js
  // In ProfileListDropMenu.js
  const { saved, setSaved } = useListSavedState(listId, userId);

  // Then later...

  <ListItemButton
    divider={isOwnProfile && deletableList ? true : false}
    onClick={() => {
      const wasSaved = saved;
      setSaved(!saved);
      setOpen(false);
      enqueueSnackbar(
        wasSaved
          ? "Watchlist unsaved from profile"
          : "Watchlist saved to profile",
        { /* other options */ }
      );
    }}
  >
    <ListItemIcon>
      <FloppyDisk size={24} />
    </ListItemIcon>
    <ListItemText
      primary={saved ? "Unsave Watchlist" : "Save Watchlist"}
      primaryTypographyProps={listItemStyling}
    />
  </ListItemButton>

```

Boom! A perfectly toggle-able save button, and the hook could be used anywhere else we want to expose controls to modify this state.  Read through it and lmk what you think

Reply:
Tasty stuff man - I really like it!!  The seamless toggle from the save to unsave option is hella slick.